DEVDOCS-5665: [update] GQL SF API end to end headless checkout flow (#188)
bigcommerce/docs: docs/api-docs/headless/end-to-end-guides/graphql-storefront.mdx

This end-to-end guide describes how to work with shopper interactions on headless storefronts. These steps let you build shopping experiences where the purchase funnel is mainly powered by the GraphQL Storefront API, from catalog browsing through checkout. The examples walk you through critical capabilities that fit many headless use cases.

<Callout type="info">
You can use the GraphQL Storefront API playground to run the example queries in this guide. To access your store's GraphQL Storefront API playground, [sign in to your store](https://login.bigcommerce.com/login), and navigate to **Settings > API > Storefront API Playground**.
</Callout>

## Prerequisites

To complete the guide, you need the following:
* [BigCommerce sandbox store](https://start.bigcommerce.com/developer-sandbox/) 
* Familiarity with creating a channel and a site

## Accessing the GraphQL Storefront API

> You can make requests to the GraphQL Storefront API in a server-to-server context. Typically, server-side API clients should use [customer impersonation tokens](/docs/rest-authentication/tokens/customer-impersonation-token) to authenticate requests to the GraphQL Storefront API, whereas client-side API clients that primarily serve static sites with no sign-in capabilities can use [storefront tokens](/docs/rest-authentication/tokens). However, server-to-server requests can use storefront tokens as well. This guide describes creating and using customer impersonation tokens to authenticate and use the GraphQL Storefront API. 

Author: ```suggestion
You can make requests to the GraphQL Storefront API in a server-to-server context. Typically, server-side API clients should use [customer impersonation tokens](/docs/rest-authentication/tokens/customer-impersonation-token) to authenticate requests to the GraphQL Storefront API. Frontend API clients that primarily serve static sites with no sign-in capabilities can use [storefront tokens](/docs/rest-authentication/tokens). However, server-to-server requests can use storefront tokens as well. This guide describes creating and using customer impersonation tokens to authenticate and use the GraphQL Storefront API. 
```

> You can make requests to the GraphQL Storefront API in a server-to-server context. Typically, server-side API clients should use [customer impersonation tokens](/docs/rest-authentication/tokens/customer-impersonation-token) to authenticate requests to the GraphQL Storefront API, whereas client-side API clients that primarily serve static sites with no sign-in capabilities can use [storefront tokens](/docs/rest-authentication/tokens). However, server-to-server requests can use storefront tokens as well. This tutorial describes creating and using customer impersonation tokens to authenticate and use the GraphQL Storefront API. 

Author: ```suggestion
You can make requests to the GraphQL Storefront API in a server-to-server context. Typically, server-side API clients should use [customer impersonation tokens](/docs/rest-authentication/tokens/customer-impersonation-token) to authenticate requests to the GraphQL Storefront API, whereas client-side API clients that primarily serve static sites with no sign-in capabilities can use [storefront tokens](/docs/rest-authentication/tokens). However, server-to-server requests can use storefront tokens as well. This guide describes creating and using customer impersonation tokens to authenticate and use the GraphQL Storefront API. 
```

### REST Management API

> * [Guide: End-to-end headless flow](/docs/storefront/headless/end-to-end-guide/rest)

Author: this one is going to change as part of this menu shift.

> * [Guide: End-to-end headless flow](/docs/storefront/headless/end-to-end-guide/rest)

Author: ```suggestion
* [Guide: End-to-end headless flow](/docs/storefront/headless/end-to-end-guides/rest-management)
```
